### PR TITLE
Add customer documentation for read/write/batch API operation performance enhancing separation

### PR DESCRIPTION
**Pull request by tapaswenipathak**
ref: https://github.com/openMF/mifos-documentation/issues/14.

cc @edcable.

## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-doc/src/docs/en/chapters/architecture/api-performance.adoc [hunk 1]
```diff
@@ -0,0 +1,115 @@
+= Introducing Performance Enhancing Read/Write/Batch API Operation Separation
```
Comment (galovics) [inline fineract-doc/src/docs/en/chapters/architecture/api-performance.adoc]: I'm not really sure why we created a new adoc file and calling it api-performance. Either it should be in the architecture doc or a brand new file which is specific to read/write/batch separation.

File: fineract-doc/src/docs/en/chapters/architecture/api-performance.adoc [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
```
Comment (galovics) [inline fineract-doc/src/docs/en/chapters/architecture/api-performance.adoc]: Also, this new file is never referenced in the index file so it doesn't appear anywhere.

**Review by galovics — CHANGES_REQUESTED** [review]

File: fineract-doc/src/docs/en/chapters/architecture/api-performance.adoc (hunk 1 above)
Comment (tapaswenipathak) [reply fineract-doc/src/docs/en/chapters/architecture/api-performance.adoc]: good point, naming the file is still open for discussion; do you think a file name as api-performance-enhancement or improvement can  have customer documentation on the best practices to work with api to get best performance? cc @edcable.

File: fineract-doc/src/docs/en/chapters/architecture/api-performance.adoc [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+* Full mode (default)
+* Read mode
+* Write mode
+* Batch mode
+
+= Setup
+
+== Environment Variables
+
+Using environment variables, the three instance types performance enhancing
```
Comment (vidakovic) [inline fineract-doc/src/docs/en/chapters/architecture/api-performance.adoc]: Please no manual line breaks... see also other parts of the doc

**Pull request closed without merging**